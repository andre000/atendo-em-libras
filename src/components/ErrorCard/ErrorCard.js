import React from 'react'
import { Card, IconText } from '../ProviderCard/ProviderCardStyles'
import errorImage from '../../assets/images/img_erro.svg'
import { refreshIcon, warningIcon } from '../../assets/icons'
import { Paragraph, Box } from 'grommet'
import styled from 'styled-components'

const RefreshIcon = styled(IconText)`
  margin-right: 8px;
`

const WarningIcon = styled.img`
  position: absolute;
  left: 45%;
  top: 45%;
`

const ErrorContainer = styled(Card)`
  width: 607px;
  height: 589px;
  text-align: center;
  vertical-align: center;
`

const ErrorImage = styled(Box)`
  position: relative;
  width: 501px;
  height: 234px;
  margin-top: 5%;
  margin-bottom: 15%;
  background: transparent url(${errorImage}) no-repeat padding-box;
`

const RefreshButton = styled.button`
  width: 209px;
  height: 55px;
  background: #5996f7 padding-box;
  border: none;
  border-radius: 17px;
  color: white;
  font-weight: 600;
  font-size: 1.4em;
  margin-top: 10%;
  cursor: pointer;

  &:hover {
    background-color: #96bfff;
  }

  &:focus {
    outline: none;
    outline-offset: 0;
  }
`

export const ErrorCard = ({ onClick }) => (
  <Box align="center" role="error">
    <ErrorContainer align="center">
      <ErrorImage aling="center">
        <WarningIcon src={warningIcon} alt="warningIcon" />
      </ErrorImage>
      <Paragraph size="large">Ops, tivemos um problema e infelizmente não conseguimos carregar a lista.</Paragraph>
      <Paragraph size="large">
        <strong>Tente novamente.</strong>
      </Paragraph>
      <RefreshButton onClick={onClick} aria-label="reload">
        <span>
          <RefreshIcon src={refreshIcon} alt="refreshIcon" />
        </span>
        Recarregar página
      </RefreshButton>
    </ErrorContainer>
  </Box>
)
