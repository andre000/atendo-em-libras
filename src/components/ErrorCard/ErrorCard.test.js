import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import { ErrorCard } from './ErrorCard'
import 'jest-styled-components'

describe('ErrorCard tests', () => {
  xit('Render correctly', () => {
    let onClick = jest.fn()
    render(<ErrorCard onClick={onClick} />)
    expect(screen.getByRole('error')).toMatchSnapshot()
  })

  it('Call reload function when button is clicked', () => {
    let onClick = jest.fn()
    render(<ErrorCard onClick={onClick} />)
    fireEvent.click(screen.getByRole('button', { name: 'reload' }))
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
