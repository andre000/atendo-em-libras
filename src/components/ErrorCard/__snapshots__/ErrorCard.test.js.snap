// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorCard tests Render correctly 1`] = `
<div
  class="StyledBox-sc-13pk1d4-0 gVSbfC"
  role="error"
>
  <div
    class="StyledBox-sc-13pk1d4-0 gVSbfC sc-AxjAm sc-fzpans hGxJex"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jJdKAu sc-fzplWN joZkpM"
    >
      <img
        alt="warningIcon"
        class="sc-fzoLsD hYtyCV"
        src="warning.svg"
      />
    </div>
    <p
      class="StyledParagraph-tbetod-0 kMGpcl"
    >
      Ops, tivemos um problema e infelizmente não conseguimos carregar a lista.
    </p>
    <p
      class="StyledParagraph-tbetod-0 kMGpcl"
    >
      <strong>
        Tente novamente.
      </strong>
    </p>
    <button
      aria-label="reload"
      class="sc-fznyAO hmvasG"
    >
      <span>
        <img
          alt="refreshIcon"
          class="sc-Axmtr sc-fzozJi jdSxEc"
          src="ic_refresh.svg"
        />
      </span>
      Recarregar página
    </button>
  </div>
</div>
`;
